### PR TITLE
Option to skip printreads and give bqsr files to the variantcalller

### PR DESCRIPTION
**Pull request by ffinfo**
This will fix BIOPET-635.

With this is becomes possible to skip printreads (this step is using a lot of diskspace) and suply the bqsr files to the variantcaller.

Please note that this is only working for GATK callers.

### CODE REVIEW

File: shiva/src/main/scala/nl/lumc/sasc/biopet/pipelines/shiva/Shiva.scala [hunk 1]
```diff
@@ -131,6 +146,41 @@ class Shiva(val parent: Configurable) extends QScript with MultisampleMappingTra
...
+        add(baseRecalibrator)
+
+        if (useAnalyzeCovariates) {
+          val baseRecalibratorAfter = BaseRecalibrator(qscript, inputBam, bqsrAfterFile.get)
+          baseRecalibratorAfter.BQSR = bqsrFile
+          add(baseRecalibratorAfter)
+          add(AnalyzeCovariates(qscript, baseRecalibrator.out, baseRecalibratorAfter.out, swapExt(dir, inputBam, ".bam", ".baserecal.pdf")))
+        }
+
+        val printReads = PrintReads(qscript, inputBam, swapExt(dir, inputBam, ".bam", ".baserecal.bam"))
```
Comment (sndrtj) [inline shiva/src/main/scala/nl/lumc/sasc/biopet/pipelines/shiva/Shiva.scala]: Lines 164-166 could be in the `usePrintreads` if block.

File: shiva/src/main/scala/nl/lumc/sasc/biopet/pipelines/shiva/Shiva.scala [hunk 2]
```diff
@@ -131,6 +146,41 @@ class Shiva(val parent: Configurable) extends QScript with MultisampleMappingTra
           add(vc)
         })
       }
+
+      /** Adds base recalibration jobs */
+      def addBaseRecalibrator(inputBam: File, dir: File, isIntermediate: Boolean, usePrintreads: Boolean): File = {
+        val baseRecalibrator = BaseRecalibrator(qscript, inputBam, bqsrFile.get) // at this point bqsrFile should exist
```
Comment (sndrtj) [inline shiva/src/main/scala/nl/lumc/sasc/biopet/pipelines/shiva/Shiva.scala]: Explicitly checking whether the `bqsrFile` is defined may not be a bad thing, especially as this is a function that could be called before it is defined at some point in the future.

**Review by bow — APPROVED** [review]

File: shiva/src/main/scala/nl/lumc/sasc/biopet/pipelines/shiva/Shiva.scala (hunk 2 above)
Comment (ffinfo) [reply shiva/src/main/scala/nl/lumc/sasc/biopet/pipelines/shiva/Shiva.scala]: Build in a other check

File: shiva/src/main/scala/nl/lumc/sasc/biopet/pipelines/shiva/Shiva.scala (hunk 1 above)
Comment (ffinfo) [reply shiva/src/main/scala/nl/lumc/sasc/biopet/pipelines/shiva/Shiva.scala]: agree, this was still there for old code ;)

**Pull request merged**